invenio_curations/services/diff.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 Graz University of Technology.
+#
+# Invenio-Curations is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Diff handling module."""
+
+from abc import abstractmethod
+
+from invenio_i18n import lazy_gettext as _
+
+from .utils import HTMLParseException, cleanup_html_tags
+
+
+class DiffException(Exception):
+    """Custom exception for diff exceptions."""
+
+    pass
+
+
+class DiffBase:
+    """Interface for classes that process diffs in context of comment creation/update."""
+
+    @abstractmethod
+    def validate_and_cleanup(self):
+        """Overrides should implement a way to validate good data and remove unwanted data."""
+        ...
+
+    @abstractmethod
+    def to_html(self, *args):
+        """Overrides should implement a way to represent the inner state of the object in HTML."""
+        ...
+
+
+class DiffElement(DiffBase):
+    """Wrapper class for a diff tuple."""
+
+    def __init__(self, diff=None):
+        """Constructs."""
+        self._diff = diff
+
+    def get_diff(self):
+        """Returns inner state."""
+        return self._diff
```

Comment:
getters aren't really a thing in python like this; instead you'd use `@property`:
```suggestion
    @property
    def diff(self):
        """Returns the current diff."""
        return self._diff
```